### ISSUE TITLE
Extra whitespace for rendered footnotes/citations

### ISSUE THREAD

**Issue by N-Wouda** [jbms/sphinx-immaterial#312]
Hello! I'm not sure if this is an _issue_, or works as desired. In any case, sections with explicit markup (for e.g. footnotes, citations) render with an additional newline between the label and the note. This can be seen in [the documentation](https://jbms.github.io/sphinx-immaterial/rst-cheatsheet/rst-cheatsheet.html#explicit-markup):
<img width="612" alt="image" src="https://github.com/jbms/sphinx-immaterial/assets/16272507/dce77a55-49c6-49c5-a84e-56f9b9169b8d">

I half hoped these to render as in [the docutils examples](https://docutils.sourceforge.io/docs/user/rst/quickref.html#explicit-markup), with the label and note on the same line:
<img width="899" alt="image" src="https://github.com/jbms/sphinx-immaterial/assets/16272507/7dc816d0-78b9-4da5-b32a-4c5c6450498b">

Other than this super minor thing I'm very happy using `sphinx-immaterial` - big kudos for such great work!

**Comment by 2bndy5** [comment]
Good catch. I would expect no new line between footnote back link and footnote content. This might be expected behavior for footnotes that are referenced more than once.

**Comment by 2bndy5** [comment]
generated HTML output,

<details><summary>docutils uses a table structure</summary>
<p>

```html
<table>
          <tbody><tr><td colspan="2"><hr>
          <!-- <tr><td colspan="2">Footnotes: -->
          </td></tr><tr><td><a name="5"><strong>[5]</strong></a></td><td> A numerical footnote.
          Note there's no colon after the <samp>]</samp>.
          </td></tr></tbody>
</table>
``` 

</p>
</details>

<details><summary>mkdocs-material uses a div->ol structure</summary>
<p>

```html
<div class="footnote">
  <hr> <ol> <li id="fn:1"> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.&nbsp;<a class="footnote-backref" href="#fnref:1" title="Jump back to footnote 1 in the text">↩</a></p> </li> <li id="fn:2"> <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla et euismod nulla. Curabitur feugiat, tortor non consequat finibus, justo purus auctor massa, nec semper lorem quam in massa.&nbsp;<a class="footnote-backref" href="#fnref:2" title="Jump back to footnote 2 in the text">↩</a></p> </li> </ol> 
</div>
``` 

</p>
</details> 

<details><summary>sphinx-immaterial uses a aside->aside structure</summary>
<p>

```html
<aside class="footnote-list brackets">
<aside class="footnote brackets" id="id6" role="doc-footnote">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id5">5</a><span class="fn-bracket">]</span></span>
<p>A numerical footnote. Note
there’s no colon after the <code class="docutils literal notranslate"><span class="pre">]</span></code>.</p>
</aside>
</aside>
``` 

</p>
</details> 

Personally, I'd prefer the footnotes be generated [as they are in mkdoc-material](https://squidfunk.github.io/mkdocs-material/reference/footnotes/), but I haven't looked at sphinx-immaterial source code yet.

**Comment by N-Wouda** [comment]
The `mkdoc-material` notes look beautiful!

**Comment by 2bndy5** [comment]
I think we need to override the `visit_footnote()` and `depart_footnote()` from docutils' HTML writer. Sphinx doesn't seem to reimplement those methods.

```py

    def visit_footnote(self, node):
        # No native HTML element: use <aside> with ARIA role
        # (html4css1 uses tables).
        # Wrap groups of footnotes for easier styling.
        label_style = self.settings.footnote_references  # brackets/superscript
        if not isinstance(node.previous_sibling(), type(node)):
            self.body.append(f'<aside class="footnote-list {label_style}">\n')
        self.body.append(self.starttag(node, 'aside',
                                       classes=[node.tagname, label_style],
                                       role="doc-footnote"))

    def depart_footnote(self, node):
        self.body.append('</aside>\n')
        if not isinstance(node.next_node(descend=False, siblings=True),
                          type(node)):
            self.body.append('</aside>\n')

```

**Comment by 2bndy5** [comment]
I did some local testing and adapted the aforementioned visit/depart methods to generate mkdocs-material's div->ol->li structure. However, the blank line persisted, so I think we need to look at CSS inherited from upstream.

**Comment by jbms** [comment]
> I did some local testing and adapted the aforementioned visit/depart methods to generate mkdocs-material's div->ol->li structure. However, the blank line persisted, so I think we need to look at CSS inherited from upstream.

The `aside` element seems like a more appropriate element, but that may indeed make it more difficult to keep the styling identical.

As far as the blank line, there is some css from the basic theme that deals with it:

https://github.com/sphinx-doc/sphinx/blob/fa290049515c38e68edda7e8c17be69b8793bb84/sphinx/themes/basic/static/basic.css_t#L616

Essentially the footnote identifier is in a `span` element and the content is in a `p` element.  The `p` element has `display: block` so it starts a new paragraph.  To fix it we can make the `span` elements float left.

**Comment by 2bndy5** [comment]
I copied over the CSS from sphinx' basic theme. I altered the paragraph indent to be slightly less than basic theme's CSS.

Rendered results for #313 can be [observed on RTD](https://sphinx-immaterial--313.org.readthedocs.build/en/313/rst-cheatsheet/rst-cheatsheet.html#explicit-markup). I think this should resolve the problem, yes?

**Comment by N-Wouda** [comment]
Thanks @2bndy5 @jbms! 🎉

**Comment by 2bndy5** [comment]
Released in [v0.11.12](https://pypi.org/project/sphinx-immaterial/0.11.12/)

**Comment by N-Wouda** [comment]
Fantastic! I'm going to pull this in immediately. Thanks again @2bndy5!
